contrib/google.golang.org/api/tree.go

```diff
@@ -0,0 +1,121 @@
+package api
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+type (
+	// A Tree is a prefix tree for matching endpoints based on http requests.
+	Tree struct {
```

Comment:
Do any of these need to be part of the public API? I'm thinking not. In which case we can make them unexported. I'd check with `godoc` to make sure nothing extra is there and it's all easy to figure out from a user perspective.

Reply:
I wasn't able to make it private because the generator script is not in the same package. I could make it internal.

Reply:
It has to be private because there is no use for this in the public API. If I'm wrong go ahead and keep it. Otherwise we should hide it. If we have to create an `internal` package for that then maybe we should.

Reply:
Let's keep it. It can't hurt, and if anyone ever wants to add some custom endpoints, they can...